### ISSUE TITLE
wasm upstream linking error

### ISSUE THREAD

**Issue by jpharvey** [emscripten-core/emscripten#9079]
[linkError.zip](https://github.com/emscripten-core/emscripten/files/3431574/linkError.zip)
Using 1.38.39-upstream i have a linking error when building our code. I have stripped down the code to the smallest i can to reproduce the error which can be built with (attached zip contains the source) :- 
```
em++ -o test.js DVec3.cpp FVec3.cpp
Assertion failed: functionIndex != INVALID_INDEX, file C:\b\s\w\ir\cache\builder\emscripten-releases\llvm-project\lld\wasm\Symbols.cpp, line 153
Stack dump:
0.      Program arguments: d:/compilers/emscripten-1.38.39-upstream/upstream/bin\wasm-ld.exe -o D:\Users\jpharvey\AppData\Local\Temp\emscripten_temp_swiz2bu0\test.wasm --allow-undefined --import-memory --import-table --lto-O0 D:\Users\jpharvey\AppData\Local\Temp\emscripten_temp_swiz2bu0\DVec3_0.o D:\Users\jpharvey\AppData\Local\Temp\emscripten_temp_swiz2bu0\FVec3_1.o d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libc.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libcompiler_rt.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libc-wasm.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libdlmalloc.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libpthreads_stub.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libcompiler_rt_wasm.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libc_rt_wasm.a -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr --export __wasm_call_ctors --export __data_end --export main --export malloc --export free --export setThrew --export __errno_location --export fflush -z stack-size=5242880 --initial-memory=16777216 --no-entry --max-memory=16777216 --global-base=1024
 #0 0x00007ff7beb2e035 (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0xae035)
 #1 0x00007ffe3680d09b (C:\WINDOWS\System32\ucrtbase.dll+0x6d09b)
 #2 0x00007ffe3680e131 (C:\WINDOWS\System32\ucrtbase.dll+0x6e131)
 #3 0x00007ffe3680fee6 (C:\WINDOWS\System32\ucrtbase.dll+0x6fee6)
 #4 0x00007ffe3680fde1 (C:\WINDOWS\System32\ucrtbase.dll+0x6fde1)
 #5 0x00007ffe3681011f (C:\WINDOWS\System32\ucrtbase.dll+0x7011f)
 #6 0x00007ff7bed0f98b (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x28f98b)
 #7 0x00007ff7bed23eeb (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x2a3eeb)
 #8 0x00007ff7bed2b07a (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x2ab07a)
 #9 0x00007ff7bed2ba95 (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x2aba95)
#10 0x00007ff7bed0e05a (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x28e05a)
#11 0x00007ff7bed0e3e7 (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x28e3e7)
#12 0x00007ff7bead4e4c (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x54e4c)
#13 0x00007ff7c0224854 (d:\compilers\emscripten-1.38.39-upstream\upstream\bin\wasm-ld.exe+0x17a4854)
#14 0x00007ffe39ee4034 (C:\WINDOWS\System32\KERNEL32.DLL+0x14034)
#15 0x00007ffe3a2c3691 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x73691)
shared:ERROR: 'd:/compilers/emscripten-1.38.39-upstream/upstream/bin\wasm-ld.exe -o D:\Users\jpharvey\AppData\Local\Temp\emscripten_temp_swiz2bu0\test.wasm --allow-undefined --import-memory --import-table --lto-O0 D:\Users\jpharvey\AppData\Local\Temp\emscripten_temp_swiz2bu0\DVec3_0.o D:\Users\jpharvey\AppData\Local\Temp\emscripten_temp_swiz2bu0\FVec3_1.o d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libc.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libcompiler_rt.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libc-wasm.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libdlmalloc.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libpthreads_stub.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libcompiler_rt_wasm.a d:\users\jpharvey\.emscripten_cache_1.38.39-upstream\wasm-obj\libc_rt_wasm.a -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr --export __wasm_call_ctors --export __data_end --export main --export malloc --export free --export setThrew --export __errno_location --export fflush -z stack-size=5242880 --initial-memory=16777216 --no-entry --max-memory=16777216 --global-base=1024' failed (2147483651)
```

**Comment by truedim** [comment]
I think it was fixed in the version 1.38.40
[https://github.com/emscripten-core/emscripten/issues/8981](https://github.com/emscripten-core/emscripten/issues/8981)

**Comment by jpharvey** [comment]
Thanks for pointing this out. Just tested 1.38.40 and get past that part of my build.
Will close this issue.
Thank you
